src/cmd/remove.nim

```diff
@@ -5,7 +5,22 @@ import ../cli/prompt, ../cli/term
 import ../modpack/files, ../modpack/install
 import ../util/flow
 
-proc paxRemove*(name: string): void =
+proc removeDependencies(manifest: var Manifest, file: ManifestFile): void =
+  ## Recursively removes dependencies of a mod
+  if len(file.dependencies) > 0:
+    for id in file.dependencies:
+      if not manifest.isInstalled(id):
+        continue
+
+      let modToRemove = manifest.getFile(id)
+
+      if not modToRemove.explicit:
+        echoInfo "Removing ", modToRemove.name.cyanFg, ".."
+        discard manifest.removeMod(id)
+        removeDependencies(manifest, modToRemove)
```

Comment:
Needs to check if the mod is isDepended.
Otherwise you can get this behavior:
Install Botania, it installs Patchouli as a dependency, then Install Psi, it requires Patchouli.
Uninstall Botania, it uninstalls Patchouli. Now Psi is without a dependency